res/controllers/Reloop-Digital-Jockey-2-IE.scripts.js

```diff
@@ -0,0 +1,878 @@
+RDJ2.JOG_SPIN_CUE_EXPONENT = 0.7; // 1.0 = linear response
+
+RDJ2.JOG_SPIN_PLAY_PEAK = 0.3; // [0.0, 1.0]
+RDJ2.JOG_SPIN_PLAY_EXPONENT = 0.7; // 1.0 = linear response
+
+RDJ2.JOG_SCRATCH_RPM = 33.333333; // 33 1/3
+RDJ2.JOG_SCRATCH_ALPHA = 0.125; // 1/8
+RDJ2.JOG_SCRATCH_BETA = RDJ2.JOG_SCRATCH_ALPHA / 32.0;
+RDJ2.JOG_SCRATCH_RAMP = true; // required for back spins
+
+// Seeking: Number of revolutions needed to seek from the beginning
+// to the end of the track.
+RDJ2.JOG_SEEK_REVOLUTIONS = 2;
+
+
+////////////////////////////////////////////////////////////////////////
+// Fixed constants                                                    //
+////////////////////////////////////////////////////////////////////////
+
+// Controller constants
+RDJ2.DECK_COUNT = 2;
+RDJ2.JOG_RESOLUTION = 148; // measured/estimated
+RDJ2.SHIFT_OFFSET = 0x1E;
+
+
+// Jog constants
+RDJ2.MIDI_JOG_DELTA_BIAS = 0x40; // center value of relative movements
+RDJ2.MIDI_JOG_DELTA_RANGE = 0x3F; // both forward (= positive) and reverse (= negative)
+
+
+// Mixxx constants
+RDJ2.MIXXX_JOG_RANGE = 3.0;
+RDJ2.MIXXX_LOOP_POSITION_UNDEFINED = -1;
+
+
+////////////////////////////////////////////////////////////////////////
+// Button/Knob map                                                    //
+////////////////////////////////////////////////////////////////////////
+
+/* This map is necessary as Reloop has designed the controller in such
+   a way that not all buttons/knobs have the same offset comparing
+   CH0 and CH1. By looking at the MIDI messages sent by the controller,
+   we can see that the hardware is designed as symmetric halves.
+
+   In other words, constant offset in hardware corresponds to symmetric
+   halves, but the controller layout is not fully symmetric.
+   (e.x. ACTIVATE 1 buttons)
+
+   Thus we need a map to preserve object oriented approach .
+
+   The below map is only for the unshifted controls as shifted ones
+   have the same handlers mapped in the xml file and the outputs
+   always refer to the unshifted controls. */
+RDJ2.BUTTONMAP_CH0_CH1 = {
+    load: [0x13, 0x4F],
+    play: [0x19, 0x55],
+    cue: [0x18, 0x54],
+    sync: [0x01, 0x3D],
+    search: [0x1A, 0x56],
+    scratch: [0x1B, 0x57],
+    fxdrywet: [0x1C, 0x58],
+    bendminus: [0x03, 0x3F],
+    bendplus: [0x04, 0x40],
+    loopin: [0x0F, 0x4B],
+    loopout: [0x10, 0x4C],
+    autoloop: [0x11, 0x4D],
+    loopactive: [0x12, 0x4E],
+    fx1assign: [0x27, 0x68],    //this is the shifted Activate 3 button
+    fx2assign: [0x2C, 0x63],    //this is the shifted FX ON button
+    highkill: [0x14, 0x50],
+};
+
+RDJ2.KNOBMAP_CH0_CH1 = {
+    loopSize: [0x28, 0x63],     //this is the shifted Dry/Wet Knob
+};
+
+
+////////////////////////////////////////////////////////////////////////
+// Logging functions                                                  //
+////////////////////////////////////////////////////////////////////////
+
+RDJ2.logDebug = function(msg) {
```

Comment:
please comment out all remaining active invocations of `logDebug`

Reply:
???
This is crazy. What do you mean?
First, commenting out debug loggigng directly in code is a really bad practice.
Second, During normal run these messages are inhibited and not printed out. You have to run Mixxx in debug mode to see them.

I really do not understand your motivation.

Reply:
It seems there is some misunderstanding here?
I'm not asking to remove that function, of course, but comment out all lines that call it, like you did with `//RDJ2.logDebug("Shifting containers");` for example. IMO it makes sense to keep lines like this (inactive) for anyone who may work on this mapping later on ([these](https://github.com/mixxxdj/mixxx/pull/4614/files#diff-43c49a90f402d9dd85e7be1428246ef8f51ca8859926c71307be14d1c57b0cc6R538-R555), too, I don't care). Though, since this mapping is to be beyond the developing and debugging phase, logDebug is of no use for users and just baggage because it's processed in the js engine (though not printed to the log during normal runs).

Reply:
Again, this is really a bad idea.
Most of the overhead of the _logDebug_ comes from the I/O and the actual printing on the console. If it is inhibited, the mere processing you mention should not be noticeable and should not impact the performance at all. Do you have any measurements that support your argument?

Reply:
You want to get this mapping merged, so implicitely you're saying "This mapping is finished.".
Please explain why you want to keep useless debug output active (while you deactivated it in other, similar important places) and why it's such a bad idea to simply comment it out (so that it can save future developers some work).

This discussion about adding `//` in a few places is of course not about the performance impact of single lines, that is negligible. It's about the general rule that debug output should only remain _if it adds value_. This is simply not the case here.

Btw, in other mappings debug output is hidden by an script-internal debug flag, and none of the more recently added, properly reviewed mappings uses print() for anything but error messages.

Reply:
@dj-ak @ronso0 Please relax a bit. You are both right: For testing it is good, but for production it should be commented out, since it slows down the overall performance and no one will look at log messages. That's like the browser's `console.log` in released websites, simply don't do it if there isn't a good reason for it like e. .g a warning for someone should do anything there.

Reply:
Thank you @alpham8 
I am relaxed, don't worry. Otherwise I wouldn't bother with a review of a controller maping I'll never use ;)

Reply:
> Btw, in other mappings debug output is hidden by an script-internal debug flag

This is how it is done here. Line 98 is protecting from printing unless the program is in debug mode:
```
    if (RDJ2.debug) {
        print("[" + RDJ2.id + " DEBUG] " + msg);
    }
```
This is also how I saw it's done in other mappings so I didn't think this might be a review problem.

Another exmple:
When you launch mixxx  with the debug log enabled:
`mixxx --logLevel debug`
You get lots of _Debug_  and _Info_ output **from the Mixxx itself** (see below).

So please explain me how Mixxx's internal debug differs from the mapping's debug. Both are invisible to the user until the user explicitly runs Mixxx with debug level enabled, **but then it is not a user anymore but a developer**.

And I understand @alpham8's analogy to websites console.log. I agree, the user should not be exposed to debug log. But then, all the Mixxx debug we see below shall be removed. This would be consistent then.

Console log:
```
mixxx --logLevel debug
Debug [Main]: SoundSourceProxy - Registering SoundSource providers
Debug [Main]: SoundSourceProxy - Registering platform and fallback SoundSource providers
Debug [Main]: SoundSourceProviderRegistry - Registering provider "FFmpeg"
Info [Main]: SoundSourceFFmpeg - Disabling untested input formats: aa, ac3, acm, act, adf, adp, ads, adx, aea, afc, aix, amr, amrnb, amrwb, anm, apc, ape, apng, aptx, aptx_hd, aqtitle, asf, asf_o, ass, ast, au, avi, avisynth, avr, avs, avs2, bethsoftvid, bfi, bin, bink, bit, bmv, bfstm, brstm, boa, c93, caf, cavsvideo, cdg, cdxl, cine, codec2, codec2raw, concat, dash, data, daud, dcstr, dfa, dhav, dirac, dnxhd, dsf, dsicin, dss, dts, dtshd, dv, dvbsub, dvbtxt, dxa, ea, ea_cdata, eac3, epaf, ffmetadata, filmstrip, fits, flac, flic, flv, live_flv, 4xm, frm, fsb, g722, g723_1, g726, g726le, g729, gdv, genh, gif, gsm, gxf, h261, h263, h264, hcom, hevc, hls, hnm, ico, idcin, idf, iff, ifv, ilbc, image2, image2pipe, alias_pix, brender_pix, ingenient, ipmovie, ircam, iss, iv8, ivf, ivr, jacosub, jv, kux, lmlm4, loas, lrc, lvf, lxf, matroska,webm, mgsts, microdvd, mjpeg, mjpeg_2000, mlp, mlv, mm, mmf, mpc, mpc8, mpeg, mpegts, mpegtsraw, mpegvideo, mpjpeg, mpl2, mpsub, msf, msnwctcp, mtaf, mtv, musx, mv, mvi, mxf, mxg, nc, nistsphere, nsp, nsv, nut, nuv, ogg, oma, paf, alaw, mulaw, vidc, f64be, f64le, f32be, f32le, s32be, s32le, s24be, s24le, s16be, s16le, s8, u32be, u32le, u24be, u24le, u16be, u16le, u8, pjs, pmp, pva, pvf, qcp, r3d, rawvideo, realtext, redspark, rl2, rm, roq, rpl, rsd, rso, rtp, rtsp, s337m, sami, sap, sbc, sbg, scc, sdp, sdr2, sds, sdx, film_cpk, ser, shn, siff, sln, smk, smjpeg, smush, sol, sox, spdif, srt, psxstr, stl, subviewer1, subviewer, sup, svag, swf, tak, tedcaptions, thp, 3dostr, tiertexseq, tmv, truehd, tta, txd, tty, ty, v210, v210x, vag, vc1, vc1test, vividas, vivo, vmd, vobsub, voc, vpk, vplayer, vqf, w64, wc3movie, webm_dash_manifest, webvtt, wsaud, wsd, wsvqa, wtv, wve, xa, xbin, xmv, xvag, xwma, yop, yuv4mpegpipe, bmp_pipe, dds_pipe, dpx_pipe, exr_pipe, gif_pipe, j2k_pipe, jpeg_pipe, jpegls_pipe, pam_pipe, pbm_pipe, pcx_pipe, pgmyuv_pipe, pgm_pipe, pictor_pipe, png_pipe, ppm_pipe, psd_pipe, qdraw_pipe, sgi_pipe, svg_pipe, sunrast_pipe, tiff_pipe, webp_pipe, xpm_pipe, xwd_pipe, libgme, libopenmpt
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "aac" for provider "FFmpeg" with priority 1 (lowest)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "aif" for provider "FFmpeg" with priority 1 (lowest)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "aiff" for provider "FFmpeg" with priority 1 (lowest)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "m4v" for provider "FFmpeg" with priority 1 (lowest)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "mov" for provider "FFmpeg" with priority 1 (lowest)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "mp4" for provider "FFmpeg" with priority 1 (lowest)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "m4a" for provider "FFmpeg" with priority 1 (lowest)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "3gp" for provider "FFmpeg" with priority 1 (lowest)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "3g2" for provider "FFmpeg" with priority 1 (lowest)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "mj2" for provider "FFmpeg" with priority 1 (lowest)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "mp3" for provider "FFmpeg" with priority 1 (lowest)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "wav" for provider "FFmpeg" with priority 1 (lowest)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "wv" for provider "FFmpeg" with priority 1 (lowest)
Debug [Main]: SoundSourceProviderRegistry - Registering provider "MAD: MPEG Audio Decoder"
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "mp3" for provider "MAD: MPEG Audio Decoder" with priority 3 (default)
Debug [Main]: SoundSourceProviderRegistry - Registering provider "MODPlug"
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "mod" for provider "MODPlug" with priority 3 (default)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "med" for provider "MODPlug" with priority 3 (default)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "okt" for provider "MODPlug" with priority 3 (default)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "s3m" for provider "MODPlug" with priority 3 (default)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "stm" for provider "MODPlug" with priority 3 (default)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "xm" for provider "MODPlug" with priority 3 (default)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "it" for provider "MODPlug" with priority 3 (default)
Debug [Main]: SoundSourceProviderRegistry - Registering provider "libsndfile"
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "aif" for provider "libsndfile" with priority 3 (default)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "aiff" for provider "libsndfile" with priority 3 (default)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "caf" for provider "libsndfile" with priority 2 (lower)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "flac" for provider "libsndfile" with priority 2 (lower)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "ogg" for provider "libsndfile" with priority 2 (lower)
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "wav" for provider "libsndfile" with priority 3 (default)
Debug [Main]: SoundSourceProxy - Registering reference SoundSource providers
Debug [Main]: SoundSourceProviderRegistry - Registering provider "Xiph.org libFLAC"
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "flac" for provider "Xiph.org libFLAC" with priority 4 (higher)
Debug [Main]: SoundSourceProviderRegistry - Registering provider "Xiph.org OggVorbis"
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "ogg" for provider "Xiph.org OggVorbis" with priority 4 (higher)
Debug [Main]: SoundSourceProviderRegistry - Registering provider "Xiph.org libopusfile"
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "opus" for provider "Xiph.org libopusfile" with priority 4 (higher)
Debug [Main]: SoundSourceProviderRegistry - Registering provider "WavPack"
Debug [Main]: SoundSourceProviderRegistry - Registering file extension "wv" for provider "WavPack" with priority 4 (higher)
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "3g2"
Info [Main]: SoundSourceProxy - 1 (lowest) : "FFmpeg"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "3gp"
Info [Main]: SoundSourceProxy - 1 (lowest) : "FFmpeg"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "aac"
Info [Main]: SoundSourceProxy - 1 (lowest) : "FFmpeg"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "aif"
Info [Main]: SoundSourceProxy - 3 (default) : "libsndfile"
Info [Main]: SoundSourceProxy - 1 (lowest) : "FFmpeg"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "aiff"
Info [Main]: SoundSourceProxy - 3 (default) : "libsndfile"
Info [Main]: SoundSourceProxy - 1 (lowest) : "FFmpeg"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "caf"
Info [Main]: SoundSourceProxy - 2 (lower) : "libsndfile"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "flac"
Info [Main]: SoundSourceProxy - 4 (higher) : "Xiph.org libFLAC"
Info [Main]: SoundSourceProxy - 2 (lower) : "libsndfile"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "it"
Info [Main]: SoundSourceProxy - 3 (default) : "MODPlug"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "m4a"
Info [Main]: SoundSourceProxy - 1 (lowest) : "FFmpeg"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "m4v"
Info [Main]: SoundSourceProxy - 1 (lowest) : "FFmpeg"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "med"
Info [Main]: SoundSourceProxy - 3 (default) : "MODPlug"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "mj2"
Info [Main]: SoundSourceProxy - 1 (lowest) : "FFmpeg"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "mod"
Info [Main]: SoundSourceProxy - 3 (default) : "MODPlug"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "mov"
Info [Main]: SoundSourceProxy - 1 (lowest) : "FFmpeg"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "mp3"
Info [Main]: SoundSourceProxy - 3 (default) : "MAD: MPEG Audio Decoder"
Info [Main]: SoundSourceProxy - 1 (lowest) : "FFmpeg"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "mp4"
Info [Main]: SoundSourceProxy - 1 (lowest) : "FFmpeg"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "ogg"
Info [Main]: SoundSourceProxy - 4 (higher) : "Xiph.org OggVorbis"
Info [Main]: SoundSourceProxy - 2 (lower) : "libsndfile"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "okt"
Info [Main]: SoundSourceProxy - 3 (default) : "MODPlug"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "opus"
Info [Main]: SoundSourceProxy - 4 (higher) : "Xiph.org libopusfile"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "s3m"
Info [Main]: SoundSourceProxy - 3 (default) : "MODPlug"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "stm"
Info [Main]: SoundSourceProxy - 3 (default) : "MODPlug"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "wav"
Info [Main]: SoundSourceProxy - 3 (default) : "libsndfile"
Info [Main]: SoundSourceProxy - 1 (lowest) : "FFmpeg"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "wv"
Info [Main]: SoundSourceProxy - 4 (higher) : "WavPack"
Info [Main]: SoundSourceProxy - 1 (lowest) : "FFmpeg"
Info [Main]: SoundSourceProxy - SoundSource providers for file extension "xm"
Info [Main]: SoundSourceProxy - 3 (default) : "MODPlug"
Debug [Main]: Mixxx 2.3.1 (git 2.3.1-68-ge80326ba68 (reloop-djaK); built on: Jan  6 2022 @ 19:23:10; flags: -pipe;-O3;-ffast-math;-funroll-loops;-fomit-frame-pointer;-mtune=generic;-Wall;-Wextra;-Woverloaded-virtual;-Wfloat-conversion;-Werror=return-type) is starting...
Debug [Main]: Compile time library versions:
Debug [Main]: Qt: 5.12.8
```

Reply:
> This is also how I saw it's done in other mappings so I didn't think this might be a review problem.

We have a lot of very old mappings with bad code quality. The debug flag is unnecessary. 

> When you launch mixxx  with the debug log enabled:
> `mixxx --logLevel debug`
> You get lots of _Debug_  and _Info_ output **from the Mixxx itself** (see below).

That is expected. Use `mixxx --controllerDebug` instead. Starting with Mixxx 2.4 we will also start using logging categories, so you can filter what messages you want to see.

Reply:
Thank you @Holzhaus for your comment. I will also appreciate your opinion on this dispute. If leaving debug messages in the mapping script is indeed not welcome, I can remove them as an alternative to commenting them out as I truly believe commenting them out in the code is a bad practice.

Reply:
I have removed debug messages that are logged at runtime. The three other that I left are printed to the console only at controller init so I hope no one will complain now ;)